pasqal_cloud/client.py

```diff
@@ -148,6 +150,43 @@ def _request(
             "HTTP Client has encountered an issue it is unable to recover from."
         )
 
+    def _request_with_pagination(
+        self,
+        method: str,
+        url: str,
+        payload: Optional[Union[Mapping, Sequence[Mapping]]] = None,
+        params: Optional[Dict[str, Any]] = None,
+    ) -> List[Dict[str, Any]]:
+        if not params:
+            params = {}
+        first_page_response = self._request(
+            method=method, url=url, payload=payload, params=params
+        )
+        all_items: List[Dict] = first_page_response["data"]
+        pagination_data = first_page_response.get("pagination")
+
+        if not pagination_data:
+            return all_items
+
+        total_nb_items = pagination_data["total"]
+        end = pagination_data["end"]
+        while end < total_nb_items:
+            params["offset"] = end
+            response: JSendPayload = self._request(
+                method=method, url=url, payload=payload, params=params
+            )
+
+            # Added to make sure the type is correctly set for mypy
+            # Should not happen
+            pagination_data = response.get("pagination")
+            if not pagination_data:
+                break
```

Comment:
If for some reason this happens and it exits the loop, we will lose the data retrieved from the last request, so maybe extend the data to all_items before the `break`